test/models/setting_test.rb

```diff
@@ -400,52 +400,48 @@ def create_setting(url)
 
     test 'encrypt URL setting when it contains password' do
       url = 'http://user:pass@example.com'
-      setting = create_setting(url)
+      attrs = { :name => "http_proxy", :value => url }
+      setting = Setting.find_or_create_by(attrs)
 
       assert setting.is_decryptable?(setting.read_attribute(:value)), 'Expected URL to be encrypted'
       assert_equal url, setting.value
     end
 
     test 'does not encrypt URL when userinfo does not contain password' do
       url = 'http://user@example.com'
-      setting = create_setting(url)
+      attrs = { :name => "http_proxy", :value => url }
+      setting = Setting.find_or_create_by(attrs)
 
       refute setting.is_decryptable?(setting.read_attribute(:value)), 'Expected URL not to be encrypted'
       assert_equal url, setting.value
     end
 
     test 'does not encrypt URL when URL does not have userinfo' do
       url = 'http://example.com'
-      setting = create_setting(url)
+      attrs = { :name => "http_proxy", :value => url }
+      setting = Setting.find_or_create_by(attrs)
 
       refute setting.is_decryptable?(setting.read_attribute(:value)), 'Expected URL not to be encrypted'
       assert_equal url, setting.value
     end
 
     test 'adds an error when URL is invalid' do
       url = 'http://example.com/hello world'
-      setting = create_setting(url)
+      attrs = { :name => "http_proxy", :value => url }
+      setting = Setting.find_or_create_by(attrs)
 
       refute setting.is_decryptable?(setting.read_attribute(:value)), 'Expected URL not to be encrypted'
-      assert_includes setting.errors[:value], "Invalid URI '#{url}'"
+      assert_includes setting.errors[:value], "Invalid HTTP(S) URL"
     end
-
-    test 'encrypt_url_if_password_present is not called if value does not start with http' do
-      setting = Setting.new(name: 'test_url', value: 'ftp://user:pass@example.com')
-
-      setting.expects(:encrypt_url_if_password_present).never
-      setting.save!
-    end
-
-    test 'encrypt_url_if_password_present is not called if value includes http but is not a URL' do
-      setting = Setting.new(name: 'test_url', value: 'my_http_proxy')
-
-      setting.expects(:encrypt_url_if_password_present).never
-      setting.save!
+    test 'no error when URL is invalid and the setting is not http_proxy' do
+      url = 'http://example.com/hello world'
+      attrs = { :name => "not_http_proxy", :value => url }
+      setting = Setting.find_or_create_by(attrs)
+      assert_empty setting.errors[:value]
```

Comment:
You can make it raise errors as an exception:
```suggestion
      Setting.find_or_create_by!(attrs)
```

In fact, it's probably best to do that everywhere.